### DISCUSSION TITLE
Pylance and Pyright CLI does not have the same output

### DISCUSSION THREAD

**Discussion by PierreSnell** [microsoft/pylance-release#3638] [Ideas]
Regarding https://github.com/microsoft/pylance-release/issues/3045#issuecomment-1317402722 and https://github.com/blakeNaccarato/boilerdata/issues/58

The only way to make pyright and pylance have the same output is to give pyright the path to pylance-extension stubs (seems obvious). 
However, this is not useful when using CI (like GitHub actions) as we need to host on our own a copy of the local stubs (microsoft/python-type-stubs doesn't seems enough).  

Is there a possibility for the pylance team to host/give access to the stubs so we can run pyright in CI and have the "If there is no issue in VsCode, there is no problem in the CI" development experience ?

**Comment by DetachHead** [comment]
you can use [basedpyright](https://github.com/detachhead/basedpyright), which is an open-source fork of pyright that allows you to use the same version of pyright in both the CLI and in vscode.

it does not have all of pylance's features unfortunately since pylance is closed source. however we have re-implemented some pylance exclusive features such as inlay hints, semantic highlighting and import suggestion code actions, and we intend to add more in the future.
